test/integration/good/code-gen/cpp.expected

```diff
@@ -13515,11 +13510,15 @@ static const std::vector<string> locations_array__ = {" (found before start of p
 
 
 template <typename T0__, typename T1__, typename T2__, typename T3__>
-Eigen::Matrix<stan::promote_args_t<T0__, T1__, T2__,
-T3__>, -1, 1>
-f(const T0__& t, const Eigen::Matrix<T1__, -1, 1>& z, const T2__& a,
-  const Eigen::Matrix<T3__, -1, 1>& b, std::ostream* pstream__) {
-  using local_scalar_t__ = stan::promote_args_t<T0__, T1__, T2__, T3__>;
+Eigen::Matrix<stan::promote_args_t<T0__, stan::value_type_t<T1__>, T2__,
+stan::value_type_t<T3__>>, -1, 1>
+f(const T0__& t, const T1__& z_arg__, const T2__& a, const T3__& b_arg__,
+  std::ostream* pstream__) {
+  using local_scalar_t__ = stan::promote_args_t<T0__,
+          stan::value_type_t<T1__>,
+          T2__,
+          stan::value_type_t<T3__>>;
+  const auto& z = to_ref(z_arg__);const auto& b = to_ref(b_arg__);
```

Comment:
One question. Its a minor thing, but I was not able to format these `to_ref()`s. If I manually added newlines, those then formatted as

```
   const auto& z = to_ref(z_arg__);
const auto& b = to_ref(b_arg__);
```
and that is even worse I guess.

Reply:
Sigh, printing newlines is some weird magic I won't pretend to understand. I thought `(list ~sep:cut pp_ref)` would fix it but apparently not.
Maybe `@[<hov>%a@]` decided the box short enough that it doesn't need line breaks. Vertical box `@[<v>%a@]` might do it.

Reply:
Oh, nice. I was like 100% I tried that, guess not. Works!